### ISSUE TITLE
D/T-Filter: Not possible to filter on e. g. D/T 5/5 only

### ISSUE THREAD

**Issue by MagpieFourtyTwo** [cgeo/cgeo#13261]
### Describe your problem!

With D/T filter it seems to be impossible to filter on exactly one D- and/or T-rating,
in contrast to e. g. Favorites filter, which can easily be set to a range of e. g. ``1000-1000`` favos.

(Ok, to be honest, it *is* possible, admittedly, even together with other (not inverted) filters, 
but only via "``Nested Filter`` => ``OR [X]`` => ``Inverse [X]`` => ``D 1-4.5 / T 1-4.5``", i. e. "``NOT[D/T: *-4.5/*-4.5``",
but this is basically impossible to comprehend for Jo(sephin)e Average, thus must be considered impossible, at least for them).

### How to reproduce?

1. Create a filter with "Difficulty / Terrain"
2. try to set it to D5 and/or T5, only ...

### Actual result after these steps?

Your filter will not only show D5 (resp. T5), but D4.5 (resp. T4.5) will be included, too.

### Expected result after these steps?

Sliders should allow to filter for exactly one D and/or T rating, i. e. min and max should be allowed to have the same value.

### Reproducible

Yes

### c:geo Version

2022.07.22-NB-4f43c1f

### System information

```
Device:
-------
- Device type: VOG-L29 (VOG-L29EEA, HUAWEI)
- Available processors: 8
- Android version: 10
- Android build: VOG-L29 10.1.0.150(C431E19R2P5)
- Screen resolution: 1080x2147px (423x841dp)
- Pixel density: 2.55
- System font scale: 1.0 / used scale: 1.0
- Sailfish OS detected: false
- Google Play services: disabled - 22.24.13 (100400-455379205)

Sensor and location:
-------
- Low power mode: active
- Compass capabilities: yes
- Rotation vector sensor: present
- Orientation sensor: present
- Magnetometer & Accelerometer sensor: present
- Direction sensor used: rotation vector

Program settings:
-------
- Settings: v9, Count:319
- Set language: en_DE (system default)
- System date format: dd/MM/y
- Time zone: GMT+02:00
- Debug mode active: no
- Last backup: 22 Jul, 00:20
- Routing mode: Bike
- Live map mode: true
- OSM multi-threading: true / threads: 4
- Map: Combined (Offline)
  - Id: cgeo.geocaching.maps.mapsforge.MapsforgeMapProvider$OfflineMultiMapSource
  - Atts: _ DE BaWü Vosges.osm.map: / Maps: (c)Christian Kernbeis & Michael Bechtold , Themes: (c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / _ DE Bayern.osm.map: / Maps: (c)Christian Kernbeis & Michael Bechtold , Themes: (c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / _ DE Hessen.osm.map: / Maps: (c)Christian Kernbeis & Michael Bechtold , Themes: (c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / _ DE North.osm.map: / Maps: (c)Christian Kernbeis & Michael Bechtold , Themes: (c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / _ DE NRW.osm.map: / Maps: (c)Christian Kernbeis & Michael Bechtold , Themes: (c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / _ DE Rheinland-Pfalz.osm.map: / Maps: (c)Christian Kernbeis & Michael Bechtold , Themes: (c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / _ DE Saarland Lux.osm.map: / Maps: (c)Christian Kernbeis & Michael Bechtold , Themes: (c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / AT CH FR Alps.osm.map: / Maps: (c)Christian Kernbeis & Michael Bechtold , Themes: (c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / BE LUX Belgium Lux.osm.map: / Maps: (c)Christian Kernbeis & Michael Bechtold , Themes: (c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / CH Switzerland.osm.map: / Maps: (c)Christian Kernbeis & Michael Bechtold , Themes: (c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / CZ Czech.osm.map: / Maps: (c)Christian Kernbeis & Michael Bechtold , Themes: (c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / DK Denmark.osm.map: / Maps: (c)Christian Kernbeis & Michael Bechtold , Themes: (c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / ES Spain Portugal.osm.map: / Maps: (c)Christian Kernbeis & Michael Bechtold , Themes: (c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / FR North.osm.map: / Maps: (c)Christian Kernbeis & Michael Bechtold , Themes: (c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / FR South.osm.map: / Maps: (c)Christian Kernbeis & Michael Bechtold , Themes: (c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / IT Italy.osm.map: / Maps: (c)Christian Kernbeis & Michael Bechtold , Themes: (c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / NL Netherlands.osm.map: / Maps: (c)Christian Kernbeis & Michael Bechtold , Themes: (c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / PL Poland.osm.map: / Maps: (c)Christian Kernbeis & Michael Bechtold , Themes: (c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / PY Pyrenees.osm.map: / Maps: (c)Christian Kernbeis & Michael Bechtold , Themes: (c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / World.map: / mapsforge-map-writer-dev-SNAPSHOT
  - Theme: Elements - shows all.xml

Filters:
-------
- Hide waypoints: -
- LIVE: None ([:inconclusive=false:advanced=true])
- OFFLINE: NOT[Last details update:--8h] ([:inconclusive=false:advanced=true]NOT(AND(stored_since:-10:28800)))
Stored Filters (#10):
- Live (Hide Found, Own & Archived): [- Live \(Hide Found, Own & Archived\):inconclusive=false:advanced=true]AND(status:owned_no:found_no:has_offline_found_log_no:exclude_archived))
- Live, but only Stored in any list: [- Live, but only Stored in any list:inconclusive=false:advanced=true]AND(status:owned_no:found_no:stored_yes:has_offline_found_log_no:exclude_archived))
- ...

Services:
-------
- Geocaching sites enabled:
   geocaching.com: Logged in (Login OK) / PREMIUM
   extremcaching.com: Logged in (Login OK)
- Geocaching.com date format: dd MMM yy
- Last successful login on geocaching.com: 2022-07-22 10-37
- Routing: external / BRouter installed: true
- Installed c:geo plugins: contacts

Permissions & paths:
-------
- Fine location permission: granted
- Write external storage permission: granted
- System internal c:geo dir: /data/user/0/cgeo.geocaching (52,6 GB free) v3 internal isDir(8 entries)
- Legacy User storage c:geo dir: /storage/emulated/0/cgeo (52,5 GB free) v3 external non-removable isDir(5 entries)
- Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (52,5 GB free) v3 external non-removable isDir(5747 entries)
- Internal theme sync (is turned ON): /data/user/0/cgeo.geocaching/MapThemeData (52,6 GB free) v3 internal isDir(12 entries)
- Public Folders: #11
  - BASE: /cgeo (User-Defined)[/cgeo[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo, Av:true, files:>=9, dirs:>=10, totalFileSize:>=1,4 GB, free space: 52,5 GB, files on device: 13278715)
  - OFFLINE_MAPS: Memory card/Locus/mapsVector (User-Defined)[Memory card/Locus/mapsVector[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/0123-4567%3ALocus%2FmapsVector::]] (Uri: content://com.android.externalstorage.documents/tree/0123-4567%3ALocus%2FmapsVector/document/0123-4567%3ALocus%2FmapsVector, Av:true, files:>=568, dirs:>=6, totalFileSize:>=31,8 GB, free space: 117,4 GB, files on device: 0)
  - OFFLINE_MAP_THEMES: Memory card/Locus/mapsVector/_themes/Elevate5 (User-Defined)[Memory card/Locus/mapsVector/_themes/Elevate5[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/0123-4567%3ALocus%2FmapsVector%2F_themes%2FElevate5::]] (Uri: content://com.android.externalstorage.documents/tree/0123-4567%3ALocus%2FmapsVector%2F_themes%2FElevate5/document/0123-4567%3ALocus%2FmapsVector%2F_themes%2FElevate5, Av:true, files:494, dirs:1, totalFileSize:3,6 MB, free space: 117,4 GB, files on device: 0)
  - LOGFILES: /cgeo/logfiles (Default)[/cgeo/logfiles[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/logfiles]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Flogfiles, Av:true, files:1, dirs:0, totalFileSize:119 B,
free space: 52,5 GB, files on device: 13278715)
  - GPX: /Download/GPX (User-Defined)[/Download/GPX[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3ADownload%2FGPX::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3ADownload%2FGPX/document/primary%3ADownload%2FGPX, Av:true, files:10, dirs:0, totalFileSize:357,2 KB, free space: 52,5 GB, files on device: 13278715)
  - BACKUP: /cgeo/backup (Default)[/cgeo/backup[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/backup]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fbackup, Av:true, files:>=14, dirs:>=5, totalFileSize:>=2,4 GB, free space: 52,5 GB, files on device: 13278715)
  - FIELD_NOTES: /cgeo/field-notes (Default)[/cgeo/field-notes[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/field-notes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Ffield-notes, Av:true, files:11, dirs:0, totalFileSize:51,9 KB, free space: 52,5 GB, files on device: 13278715)
  - SPOILER_IMAGES: /cgeo/GeocachePhotos (Default)[/cgeo/GeocachePhotos[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/GeocachePhotos]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2FGeocachePhotos, Av:true, files:>=4, dirs:>=12, totalFileSize:>=269,2 KB, free space: 52,5 GB, files on device: 13278715)
  - ROUTING_BASE: /cgeo/routing (Default)[/cgeo/routing[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/routing]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Frouting, Av:true, files:11, dirs:0, totalFileSize:109,9 KB, free space: 52,5 GB, files on device: 13278715)
  - ROUTING_TILES: Memory card/Android/media/btools.routingapp/brouter/segments4 (User-Defined)[Memory card/Android/media/btools.routingapp/brouter/segments4[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/0123-4567%3AAndroid%2Fmedia%2Fbtools.routingapp%2Fbrouter%2Fsegments4::]] (Uri: content://com.android.externalstorage.documents/tree/0123-4567%3AAndroid%2Fmedia%2Fbtools.routingapp%2Fbrouter%2Fsegments4/document/0123-4567%3AAndroid%2Fmedia%2Fbtools.routingapp%2Fbrouter%2Fsegments4, Av:true, files:39, dirs:0, totalFileSize:1,7 GB, free space: 117,4 GB, files on device: 0)
  - TEST_FOLDER: [Legacy]/data/user/0/cgeo.geocaching/files/unittest (Default)[/data/user/0/cgeo.geocaching/files/unittest[FILE#1:p-file:///data/user/0/cgeo.geocaching/files::/unittest]] (Uri: file:///data/user/0/cgeo.geocaching/files/unittest, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 52,6 GB, files on device: -1)
- Map render theme path: Elements - shows all.xml
- PersistedDocumentUris: #1
- UNUSED_TRACK: null
- Persisted Uri Permissions: #5
  - content://com.android.externalstorage.documents/tree/0123-4567%3AAndroid%2Fmedia%2Fbtools.routingapp%2Fbrouter%2Fsegments4 (25 Dec 2021, 21:59):RW
  - content://com.android.externalstorage.documents/tree/primary%3ADownload%2FGPX (24 May 2021, 00:17):RW
  - content://com.android.externalstorage.documents/tree/0123-4567%3ALocus%2FmapsVector%2F_themes%2FElevate5 (1 May, 03:56):RW
  - content://com.android.externalstorage.documents/tree/primary%3Acgeo (20 May 2021, 23:37):RW
  - content://com.android.externalstorage.documents/tree/0123-4567%3ALocus%2FmapsVector (20 May 2021, 23:37):RW
- Database: /data/user/0/cgeo.geocaching/databases/data (v98, Size:511,7 MB) on system internal storage
--- End of system information ---
```

**Comment by eddiemuc** [comment]
I can reproduce, and of course this is a clear bug: filtering for single D or T values should be possible (and was in the past).

@MagpieFourtyTwo I could only reproduce bug in "advanced" filter mode, filtering for single D/T values seem to work in "basic" filter mode. Can you confirm this?

**Comment by fm-sys** [comment]
> I could only reproduce bug in "advanced" filter mode, filtering for single D/T values seem to work in "basic" filter mode. Can you confirm this?

At least for me, it happens independently of basic/advanced

**Comment by MagpieFourtyTwo** [comment]
Same over here: Behavior is independent of filter mode.

**Comment by MagpieFourtyTwo** [comment]
Maybe it's OS version or device dependent ... thus I will add sys info to first post.

BTW, as I just stumbled upon this, again (and don't want to open a issue, as it may just be a local problem): Trying to update a list of 2000 caches, which runs in background (can this be deactivated in any way?), but the update is *horribly* slow: After 10 hours just 800 caches are updated, which makes about a minute each - has this to be considered "normal"? I would just call it "unusable" ... :/

**Comment by fm-sys** [comment]
> BTW, as I just stumbled upon...

Please, stay focused... It makes it really complicated for us devs to keep track on issues if so much off-topic is going on. If you want to see issues solved please open a new issue. If the issue templates doesn't seem to fit its also perfectly fine to open a blank issue and describe your discovery in a few sentences...

**Comment by fm-sys** [comment]
The discussion should be continued at  #13262

**Comment by eddiemuc** [comment]
For clarification: 
* is this issue about the unability to **graphically select** only one D/T element (a phenomen which I can only reproduce in the emulator and not on my physical device)?
* or is it about n**on-correct filtering** when only one value was actually selected (a phenomen which I can't reproduce on emulator, and frankly I am not sure whether it really occured on my physical device)?

 For the first one I will supply a PR in some minutes

**Comment by MagpieFourtyTwo** [comment]
It's "about the unability to graphically select only one D/T element".

It's not possible to let the sliders touch resp. cover each other, there is always at least half a point distance between them. In "Favorites" , "Logs Count"  and "Distance" filter it works like I would expect it for "D/T", too.

"Rating" , BTW, which possibly uses the same function like "D/T", as it has predefined values, too (in contradiction to the others, which just use a pure integer scale), has the same problem: not possible to filter for just one value.

Last one with sliders is "Last Details Update", which behaves like "D/T", too (over here), but although it makes pretty sense for this one, I would prefer a finer granularity (which may need an own issue?).

**Comment by eddiemuc** [comment]
Thanks for explaining, this should be fixed with the #13319

**Comment by moving-bits** [comment]
This has been fixed meanwhile, fix is available in current release 2022.08.21, so let's close here.
